api/filters.py

```diff
@@ -23,3 +23,34 @@ def filter_with_multiple_query_values(self, queryset, name, value):
     class Meta:
         model = Asset
         fields = ['asset_type', 'model_number', 'email']
+
+
+class UserFilter(filters.FilterSet):
+    cohort = filters.CharFilter(
+        field_name='cohort',
+        lookup_expr='iexact',)
+
+    email = filters.CharFilter(
+        field_name='email',
+        lookup_expr='istartswith',
+        method='filter_by_email_or_allocated_asset_count',)
```

Comment:
Specifying a [method](https://django-filter.readthedocs.io/en/latest/ref/filters.html#method) argument on a field, means you want to resolve the filtering on the queryset yourself rather than leaving `django-filter` to do it for you. It defeats the purpose, since the `field_name` and `lookup_expr` arguments are sufficient to return the filtered result we would like.